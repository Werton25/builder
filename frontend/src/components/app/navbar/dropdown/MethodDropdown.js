import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';
import {Dropdown} from 'bootstrap.native';
import {encodeUri} from "../../../../utils";

export default class MethodDropdown extends React.Component {
  componentDidMount() {
    document.getElementsByName('method').forEach((dropdown) => new Dropdown(dropdown, {persist: true}));
  }

  render() {
    const {service} = this.props;

    const methodLinks = service.methods.map((method, i) => (
      <li key={i} role='presentation'>
        <Link role='menuitem' to={encodeUri(`/${service.name}/${method.name}`)}>{method.name}</Link>
      </li>
    ));

    return (
      <li role='presentation'>
        <a name='method' role='button' data-toggle='dropdown'>
          {service.name}
          <span className='caret'/>
        </a>
        <ul className='dropdown-menu pull-right' role='menu'>
          {methodLinks}
        </ul>
      </li>
    );

  }
}

MethodDropdown.propTypes = {
  service: PropTypes.object.isRequired
};
