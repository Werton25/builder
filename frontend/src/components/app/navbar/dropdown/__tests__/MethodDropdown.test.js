import React from "react";
import {MemoryRouter} from "react-router-dom";
import MethodDropdown from "../MethodDropdown";
import renderer from 'react-test-renderer';

describe('MethodDropdown tests', () => {
  it('MethodDropdown renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter initialEntries={['/']}>
          <MethodDropdown service={{methods: []}}/>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
