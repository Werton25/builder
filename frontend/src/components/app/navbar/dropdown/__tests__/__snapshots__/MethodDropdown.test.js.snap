// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MethodDropdown tests MethodDropdown renders correctly 1`] = `
<li
  role="presentation"
>
  <a
    data-toggle="dropdown"
    name="method"
    role="button"
  >
    <span
      className="caret"
    />
  </a>
  <ul
    className="dropdown-menu pull-right"
    role="menu"
  />
</li>
`;
